### ISSUE TITLE
ModelAdmin can't search with Datetime in searchable_fields

### ISSUE THREAD

**Issue by chillu** [silverstripe/silverstripe-framework#5031]
Verified in 3.2 and 3.3.  `SS_Datetime->scaffoldFormField()` returns a `DatetimeField`, which sends its values as separate `date` and `time` query params. The `PartialMatchFilter` doesn't think this field is empty (it's an array with keys), which means it includes it in the query - which will always cancel out any results.

Suggested fix: Implement `DateFilter extends SearchFilter` which can deal with the separate keys, or fix `ModelAdmin->getList()` to use `FormField->getValue()` rather than request params.

Example query:
```
SELECT DISTINCT count(DISTINCT "Contract"."ID") AS "Count" FROM "Contract" WHERE (("Contract"."MyDate" LIKE ?) OR ("Contract"."MyDate" LIKE ?))
```

Example code:
```php
<?php
class Contract extends DataObject {

	private static $db = array(
		'Number' => 'Varchar',
		'MyDate' => 'SS_Datetime'
	);

	private static $summary_fields = array(
		'Number' => 'Number',
		'MyDate' => 'My Date'
	);

}

<?php
class ContractAdmin extends ModelAdmin {

    private static $managed_models = array(
		'Contract',
	);

    private static $url_segment = 'contract';

    private static $menu_title = 'Contracts';

}
```

**Comment by tractorcow** [comment]
Would it be simpler if datetimefield simply posted back a single value? jquey datepicker has a time select extension that could be used.

**Comment by chillu** [comment]
Yeah that's an API change on `DatetimeField` though, and we're trying to fix this bug in 3.x. Which of the two options I've outlined above do you prefer?

**Comment by tractorcow** [comment]
> ModelAdmin->getList() to use FormField->getValue() rather than request params.

Sounds the most robust. ;d

**Comment by rasstislav** [comment]
This issue is also present in SS 4. Any workaround?
Thanks
